--- pod/podfile/views.py	2021-04-11 10:30:37.900031635 +0200
+++ /root/sauvegarde_pofile_views.py	2021-04-10 21:17:50.845633667 +0200
@@ -69,9 +69,9 @@
     user_home_folder = get_object_or_404(
         UserFolder, name="home", owner=request.user)
 
-    share_folder = UserFolder.objects.filter(
-        access_groups=request.user.owner.accessgroup_set.all()).exclude(
-            owner=request.user).order_by('owner', 'id')
+#    share_folder = UserFolder.objects.filter(
+#        groups__in=request.user.groups.all()
+#    ).exclude(owner=request.user).order_by('owner', 'id')
 
     share_folder_user = UserFolder.objects.filter(
         users=request.user).exclude(
@@ -87,7 +87,7 @@
                   {
                       'user_home_folder': user_home_folder,
                       'user_folder': [],
-                      'share_folder': share_folder,
+                      'share_folder': share_folder_user,
                       'share_folder_user': share_folder_user,
                       'current_session_folder': current_session_folder,
                       'form_file': CustomFileModelForm(),
@@ -103,10 +103,7 @@
            Q(owner=request.user, name=request.session.get(
               'current_session_folder', "home")) | Q(
                  users=request.user, name=request.session.get(
-                    'current_session_folder', "home")) | Q(
-                 access_groups=request.user.owner.accessgroup_set.all(
-                 ), name=request.session.get(
-                         'current_session_folder', "home")))
+                    'current_session_folder', "home")))
     except ObjectDoesNotExist:
         if(request.user.is_superuser):
             try:
@@ -126,17 +123,17 @@
 @csrf_protect
 @staff_member_required(redirect_field_name='referrer')
 def get_folder_files(request, id, type=None):
+
     if type is None:
         type = request.GET.get('type', None)
     folder = get_object_or_404(UserFolder, id=id)
-
     if (request.user != folder.owner
-            and not (folder.access_groups.filter(
-                code_name__in=[
+            and not request.user.groups.filter(
+                name__in=[
                     name[0]
-                    for name in request.user.owner.accessgroup_set.values_list(
-                        'code_name')
-                ]).exists())
+                    for name in folder.groups.values_list('name')
+                ]
+            ).exists()
             and not (
                 request.user.is_superuser or request.user.has_perm(
                     "podfile.change_userfolder")) and not (
@@ -171,7 +168,7 @@
     ).exclude(owner=request.user, name="home")
 
     share_folder = UserFolder.objects.filter(
-        access_groups__in=request.user.owner.accessgroup_set.all()
+        groups__in=request.user.groups.all()
     ).exclude(owner=request.user).order_by('owner', 'id')
 
     share_folder_user = UserFolder.objects.filter(
@@ -540,12 +537,12 @@
     else:
         reqfile = get_object_or_404(CustomFileModel, id=id)
     if (request.user != reqfile.folder.owner
-            and not reqfile.folder.access_groups.filter(
-                code_name__in=[
+            and not request.user.groups.filter(
+                name__in=[
                     name[0]
-                    for name in request.user.owner.accessgroup_set.values_list(
-                        'code_name')
-                ]).exists()
+                    for name in reqfile.folder.groups.values_list('name')
+                ]
+            ).exists()
             and not (request.user.is_superuser or request.user.has_perm(
                     "podfile.change_customfilemodel") or request.user.has_perm(
                     "podfile.change_customimagemodel") or (
